cmd/restic/integration_test.go

```diff
@@ -1283,6 +1283,103 @@ func TestPruneWithDamagedRepository(t *testing.T) {
 	t.Log(err)
 }
 
+func TestIndexMissing(t *testing.T) {
+	env, cleanup := withTestEnvironment(t)
+	defer cleanup()
+
+	datafile := filepath.Join("testdata", "repo-index-missing.tar.gz")
+	rtest.SetupTarTestFixture(t, env.base, datafile)
+
+	rtest.Assert(t, runCheck(CheckOptions{}, env.gopts, nil) != nil,
+		"check should have reported an error")
+	rtest.Assert(t, runPrune(env.gopts) != nil,
+		"prune should have reported an error")
+}
+
+func TestIndexMissingBlob(t *testing.T) {
```

Comment:
The other tests are also identical except for the tar-file, the check parameters and expected check result. Please merge these too.

Reply:
What does `IndexMissingBlob` check? That a single blob is not contained in the index, but still exists in the repository?

Reply:
Yes - `repo-index-missing-blob.tar.gz` is a repo where the packs are fine but a single (used) blob is missing from the index. With current `prune` this will be simply fixed (as `prune` always starts with the `rebuild-index` step), but after #2718 this will result in a prune error, which has to be explicitly fixed, e.g. by `rebuild-index`.

Reply:
The tests are now all merged.